tests/unit/test_mac_frame.cpp

```diff
@@ -355,20 +379,28 @@ void TestMacHeader(void)
             break;
         }
 
-        frame.InitMacHeader(Mac::Frame::kTypeData, testCase.mVersion, addresses, panIds, testCase.mSecurity,
+        frame.InitMacHeader(testCase.mType, testCase.mVersion, addresses, panIds, testCase.mSecurity,
                             testCase.mKeyIdMode, testCase.mSuppressSequence);
 
         VerifyOrQuit(frame.GetHeaderLength() == testCase.mHeaderLength);
         VerifyOrQuit(frame.GetFooterLength() == testCase.mFooterLength);
         VerifyOrQuit(frame.GetLength() == testCase.mHeaderLength + testCase.mFooterLength);
 
-        VerifyOrQuit(frame.GetType() == Mac::Frame::kTypeData);
+        VerifyOrQuit(frame.GetType() == testCase.mType);
         VerifyOrQuit(!frame.IsAck());
         VerifyOrQuit(frame.GetVersion() == testCase.mVersion);
         VerifyOrQuit(frame.GetSecurityEnabled() == (testCase.mSecurity != kNoSec));
         VerifyOrQuit(!frame.GetFramePending());
         VerifyOrQuit(!frame.IsIePresent());
-        VerifyOrQuit(frame.GetAckRequest() == (testCase.mDstAddrType != kNoneAddr));
+
+        if (testCase.mType == kTypeMultipurpose)
+        {
+            VerifyOrQuit(frame.GetAckRequest() == false);
```

Comment:
I know some code/style-checkers emit warnings when bools are compared using  `== false` and `== true` (which is technically correct). Suggest to not use it :)
```suggestion
            VerifyOrQuit(!frame.GetAckRequest());
```